files/image_config/logrotate/logrotate.d/rsyslog

```diff
@@ -85,7 +85,15 @@
     endscript
     postrotate
         if [ $(echo $1 | grep -c "/var/log/swss/") -gt 0 ]; then
-            pgrep -x orchagent | xargs /bin/kill -HUP 2>/dev/null || true
+            # for multi asic platforms, there are multiple orchagents
+            # send the SIGHUP only to the orchagent the which needs log file rotation
+            if [ $(pgrep -x orchagent | wc -l) -gt 0 ]; then
+                log_file=$1
+                log_file_name=${log_file#/var/log/swss/}
+                pgrep -xa orchagent | grep $log_file_name | awk '{ print $1; }' | xargs /bin/kill -HUP 2>/dev/null || true
```

Comment:
what does the pgrep -xa orchagent looks like on a multi-asic platform? can you provide output?

Reply:
```
admin@sonic:/var/log/swss$ pgrep -xa orchagent
11710 /usr/bin/orchagent -d /var/log/swss -b 8192 -i 03:00.0 -f swss.asic0.rec -j sairedis.asic0.rec -m 00:be:75:3a:ef:50
12159 /usr/bin/orchagent -d /var/log/swss -b 8192 -i 04:00.0 -f swss.asic1.rec -j sairedis.asic1.rec -m 00:be:75:3a:ef:50
12168 /usr/bin/orchagent -d /var/log/swss -b 8192 -i 08:00.0 -f swss.asic4.rec -j sairedis.asic4.rec -m 02:42:f0:7f:01:06
12169 /usr/bin/orchagent -d /var/log/swss -b 8192 -i 09:00.0 -f swss.asic5.rec -j sairedis.asic5.rec -m 02:42:f0:7f:01:02
12205 /usr/bin/orchagent -d /var/log/swss -b 8192 -i 05:00.0 -f swss.asic2.rec -j sairedis.asic2.rec -m 00:be:75:3a:ef:50
12236 /usr/bin/orchagent -d /var/log/swss -b 8192 -i 06:00.0 -f swss.asic3.rec -j sairedis.asic3.rec -m 00:be:75:3a:ef:50
```